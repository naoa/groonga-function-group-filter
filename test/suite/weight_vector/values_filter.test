plugin_register functions/group_filter

table_create Tags TABLE_PAT_KEY ShortText

table_create Memos TABLE_HASH_KEY ShortText
column_create Memos tags COLUMN_VECTOR|WITH_WEIGHT Tags

column_create Tags memos_tags COLUMN_INDEX|WITH_WEIGHT Memos tags

load --table Memos
[
{
  "_key": "Groonga is fast",
  "tags": {
    "groonga": 100
  }
},
{
  "_key": "Mroonga is also fast",
  "tags": {
    "mroonga": 100,
    "groonga": 10
  }
},
{
  "_key": "Ruby is an object oriented script language",
  "tags": {
    "ruby": 120
  }
}
]

select Memos \
  --filter 'values_filter(tags, "mroonga ruby")' \
  --output_columns _key,_score,tags,#group_tags \
  --drilldowns[sum_tag].keys #group_tags \
  --drilldowns[sum_tag].calc_types SUM \
  --drilldowns[sum_tag].calc_target tags \
  --drilldowns[sum_tag].output_columns _key,_nsubrecs,_sum
