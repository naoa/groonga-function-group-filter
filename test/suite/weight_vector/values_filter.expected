plugin_register functions/group_filter
[[0,0.0,0.0],true]
table_create Tags TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
table_create Memos TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create Memos tags COLUMN_VECTOR|WITH_WEIGHT Tags
[[0,0.0,0.0],true]
column_create Tags memos_tags COLUMN_INDEX|WITH_WEIGHT Memos tags
[[0,0.0,0.0],true]
load --table Memos
[
{
  "_key": "Groonga is fast",
  "tags": {
    "groonga": 100
  }
},
{
  "_key": "Mroonga is also fast",
  "tags": {
    "mroonga": 100,
    "groonga": 10
  }
},
{
  "_key": "Ruby is an object oriented script language",
  "tags": {
    "ruby": 120
  }
}
]
[[0,0.0,0.0],3]
select Memos   --filter 'values_filter(tags, "mroonga ruby")'   --output_columns _key,_score,tags,#group_tags   --drilldowns[sum_tag].keys #group_tags   --drilldowns[sum_tag].calc_types SUM   --drilldowns[sum_tag].calc_target tags   --drilldowns[sum_tag].output_columns _key,_nsubrecs,_sum
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        2
      ],
      [
        [
          "_key",
          "ShortText"
        ],
        [
          "_score",
          "Int32"
        ],
        [
          "tags",
          "Tags"
        ],
        [
          "#group_tags",
          "Tags"
        ]
      ],
      [
        "Ruby is an object oriented script language",
        121,
        {
          "ruby": 120
        },
        {
          "ruby": 120
        }
      ],
      [
        "Mroonga is also fast",
        101,
        {
          "mroonga": 100,
          "groonga": 10
        },
        {
          "mroonga": 100
        }
      ]
    ],
    {
      "sum_tag": [
        [
          2
        ],
        [
          [
            "_key",
            "ShortText"
          ],
          [
            "_nsubrecs",
            "Int32"
          ],
          [
            "_sum",
            "Float"
          ]
        ],
        [
          "ruby",
          1,
          120.0
        ],
        [
          "mroonga",
          1,
          110.0
        ]
      ]
    }
  ]
]
